bridges/snowbridge/primitives/core/src/transact.rs

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+
+use crate::{AssetMetadata, Vec};
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_core::H160;
+use sp_runtime::RuntimeDebug;
+use xcm::prelude::Location;
+
+#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct TransactInfo {
+	pub kind: TransactKind,
+	pub params: Vec<u8>,
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum TransactKind {
+	RegisterToken,
+	RegisterAgent,
+	CallContract,
+}
```

Comment:
An custom data modal which can be encoded to `Transact` instruction and sent through xcm.